types/shared.d.ts

```diff
@@ -374,13 +374,40 @@ export interface IStripeConnectInitParams {
   locale?: string;
 }
 
+type ConnectHTMLElementRecordFallback = {
+  [key in string]: HTMLElement | null;
+}
+type ConnectHTMLElementRecordBase = {
+  [tagName in ConnectElementTagName]: HTMLElement;
+};
+
+interface ConnectHTMLElementRecord extends ConnectHTMLElementRecordBase, ConnectHTMLElementRecordFallback {
+  "payment-details": HTMLElement & {
+    setPayment: (payment: string | undefined) => void;
+    setOnClose: (listener: () => void) => void;
+  };
+  "account-onboarding": HTMLElement & {
+    setFullTermsOfServiceUrl: (termOfServiceUrl: string | undefined) => void;
+    setRecipientTermsOfServiceUrl: (
+      recipientTermsOfServiceUrl: string | undefined
+    ) => void;
+    setPrivacyPolicyUrl: (privacyPolicyUrl: string | undefined) => void;
+    setSkipTermsOfServiceCollection: (
+      skipTermsOfServiceCollection: boolean | undefined
+    ) => void;
+    setOnExit: (listener: () => void) => void;
```

Comment:
This should allow to set an undefined listener, so that you can clear it if you want.  Similarly this method should work when I pass undefined: https://livegrep.corp.stripe.com/view/stripe-internal/pay-server/frontend/connect/submerchant-surfaces/src/connect/utils/wrapComponentAsCustomElement.ts#L70, however it looks like right now it does not... (it would try to add an undefined event listener) we should modify it.

Reply:
also made change for this in pay-server, will make a PR once build passes